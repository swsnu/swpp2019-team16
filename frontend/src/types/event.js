export const GROUP_CREATED_EVENT = 'event.group_created';
export const GROUP_DRIVER_UPDATED_EVENT = 'event.group_driver_updated';
export const RIDER_ON_TAXI_EVENT = 'event.rider_on_taxi';
export const DRIVER_GO_TAXI_EVENT = 'event.driver_go_taxi';
export const GROUP_COST_UPDATED_EVENT = 'event.group_cost_updated';
