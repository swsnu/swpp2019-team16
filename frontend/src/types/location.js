import PropTypes from 'prop-types';

export const LocationPropTypes = PropTypes.shape({
  name: PropTypes.string.isRequired,
  coordinate: PropTypes.shape({
    lat: PropTypes.number.isRequired,
    lng: PropTypes.number.isRequired,
  }).isRequired,
});

export class Location {
  /*
   * args
   *   name: name is name of the location
   *   location: location is object type which has @lat @lng as key
   * */
  constructor({ name, coordinate }) {
    this.name = name;
    this.coordinate = coordinate;
  }
}

export const fromLocationList = [
  new Location({
    name: 'SNU station',
    coordinate: {
      lat: 37.480126,
      lng: 126.952436,
    },
  }),
  new Location({
    name: 'Nakseongdae station',
    coordinate: {
      lat: 37.477023,
      lng: 126.961957,
    },
  }),
  new Location({
    name: 'Nokdu Street',
    coordinate: {
      lat: 37.470463,
      lng: 126.940765,
    },
  }),
];

export const toLocationList = [
  new Location({
    name: '301 building',
    coordinate: {
      lat: 37.448475,
      lng: 126.95203,
    },
  }),
  new Location({
    name: 'Student Center',
    coordinate: {
      lat: 37.46046,
      lng: 126.949621,
    },
  }),
  new Location({
    name: 'Dormitory three-way',
    coordinate: {
      lat: 37.461211,
      lng: 126.956347,
    },
  }),
];
