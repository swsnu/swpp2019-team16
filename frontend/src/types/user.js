import PropTypes from 'prop-types';

export const UserPropsTypes = PropTypes.shape({
  // rider/driver id
  id: PropTypes.number.isRequired,
  status: PropTypes.oneOf(['IDLE', 'ON_TAXI', 'DRIVING']).isRequired,
  // group id
  group: PropTypes.number,
  user: PropTypes.shape({
    // user id
    id: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired,
    user_type: PropTypes.oneOf(['rider', 'driver']).isRequired,
    vehicle: PropTypes.shape({
      car_type: PropTypes.string.isRequired,
      plate_no: PropTypes.string.isRequired,
    }),
  }).isRequired,
});
