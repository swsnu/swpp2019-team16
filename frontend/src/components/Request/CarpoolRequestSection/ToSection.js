import React from 'react';
import { makeStyles } from '@material-ui/styles';
import Checkbox from '../../common/Checkbox';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';
import Map from '../../common/Map';
import MapPin from '../../common/MapPin';
import { LocationPropTypes } from '../../../types/location';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
  },
  title: {
    marginBottom: '30px',
  },
  content: {
    marginBottom: '25px',
  },
  map: {
    width: '100%',
    flex: 1,
    marginBottom: '25px',
  },
  noLocationSelected: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '400px',
  },
});

ToSection.propsTypes = {
  toList: PropTypes.arrayOf(LocationPropTypes).isRequired,
  selectedTo: LocationPropTypes,
  onClickTo: PropTypes.func.isRequired,
};

function ToSection({ toList, selectedTo, onClickTo }) {
  const styles = useStyles();
  return (
    <div className={styles.root}>
      <div className={styles.title}>
        <Typography variant={'h4'}>Where do you want to go?</Typography>
      </div>
      <div className={styles.content}>
        {toList &&
          toList.map(location => (
            <Checkbox
              color={'secondary'}
              key={location.name}
              name={'from'}
              value={location.name}
              onClick={() =>
                selectedTo === location.name
                  ? onClickTo(null)
                  : onClickTo(location)
              }
              checked={selectedTo !== null && selectedTo.name === location.name}
            />
          ))}
      </div>
      <div className={styles.map}>
        {selectedTo === null ? (
          <div className={styles.noLocationSelected}>
            <Typography variant={'h5'}>No location selected.</Typography>
          </div>
        ) : (
          <div>
            <Map
              center={{
                lat: selectedTo.coordinate.lat,
                lng: selectedTo.coordinate.lng,
              }}
              width={'100%'}
              height={'400px'}
            >
              <MapPin
                lat={selectedTo.coordinate.lat}
                lng={selectedTo.coordinate.lng}
              />
            </Map>
          </div>
        )}
      </div>
    </div>
  );
}

export default ToSection;
