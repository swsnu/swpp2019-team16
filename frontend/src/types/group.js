import PropTypes from 'prop-types';
import { UserPropsTypes } from './user';

export const GroupPropTypes = PropTypes.shape({
  groupId: PropTypes.number.isRequired,
  driver: UserPropsTypes,
  riders: PropTypes.arrayOf(UserPropsTypes),
  from: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
});
