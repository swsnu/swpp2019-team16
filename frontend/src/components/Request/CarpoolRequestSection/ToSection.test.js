import React from 'react';
import { render } from '@testing-library/react';
import ToSection from './ToSection';
import { toLocationList } from '../../../types/location';

describe('<ToSection />', () => {
  it('should match with snapshot - default', async () => {
    const { container } = render(
      <ToSection
        toList={toLocationList}
        onClickTo={() => {}}
        selectedTo={toLocationList[0]}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match with snapshot - when no selected location', async () => {
    const { container } = render(
      <ToSection
        toList={toLocationList}
        onClickTo={() => {}}
        selectedTo={null}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
