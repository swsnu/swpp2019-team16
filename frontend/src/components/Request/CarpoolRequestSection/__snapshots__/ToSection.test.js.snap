// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToSection /> should match with snapshot - default 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-title-2"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Where do you want to go?
      </h4>
    </div>
    <div
      class="makeStyles-content-3"
    >
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="301 building"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-48 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-49 Mui-checked MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-51"
                data-indeterminate="false"
                name="from"
                type="checkbox"
                value="301 building"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            301 building
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="Student Center"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-48 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-51"
                data-indeterminate="false"
                name="from"
                type="checkbox"
                value="Student Center"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Student Center
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="Dormitory three-way"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-48 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-51"
                data-indeterminate="false"
                name="from"
                type="checkbox"
                value="Dormitory three-way"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Dormitory three-way
          </span>
        </label>
      </div>
    </div>
    <div
      class="makeStyles-map-4"
    >
      <div>
        <div
          class="makeStyles-root-73 makeStyles-root-74"
        >
          <div
            style="width: 100%; height: 100%; margin: 0px; padding: 0px; position: relative;"
          >
            <div
              style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
            />
            <div
              style="width: 50%; height: 50%; left: 50%; top: 50%; margin: 0px; padding: 0px; position: absolute;"
            >
              <div
                style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
              >
                <div
                  style="width: 0px; height: 0px; left: 0px; top: 0px; background-color: transparent; position: absolute;"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root makeStyles-root-75"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root makeStyles-icon-76 MuiSvgIcon-colorSecondary"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2c-4.97 0-9 4.03-9 9 0 4.17 2.84 7.67 6.69 8.69L12 22l2.31-2.31C18.16 18.67 21 15.17 21 11c0-4.97-4.03-9-9-9zm0 2c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.3c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ToSection /> should match with snapshot - when no selected location 1`] = `
<div>
  <div
    class="makeStyles-root-87"
  >
    <div
      class="makeStyles-title-88"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Where do you want to go?
      </h4>
    </div>
    <div
      class="makeStyles-content-89"
    >
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="301 building"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-134 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-137"
                data-indeterminate="false"
                name="from"
                type="checkbox"
                value="301 building"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            301 building
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="Student Center"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-134 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-137"
                data-indeterminate="false"
                name="from"
                type="checkbox"
                value="Student Center"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Student Center
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="Dormitory three-way"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-134 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-137"
                data-indeterminate="false"
                name="from"
                type="checkbox"
                value="Dormitory three-way"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Dormitory three-way
          </span>
        </label>
      </div>
    </div>
    <div
      class="makeStyles-map-90"
    >
      <div
        class="makeStyles-noLocationSelected-91"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          No location selected.
        </h5>
      </div>
    </div>
  </div>
</div>
`;
